docs/guides/web/nginx-mainline.md

```diff
@@ -0,0 +1,282 @@
+---
+title: Nginx
+author: Ezequiel Bruni
+contributors: Steven Spencer (most likely)
+---
+# How to Install the Latest Nginx on Rocky Linux
+
+## Introduction
+
+To give credit where credit is due, I came up with exactly none of this. In particular, this guide is based heavily on [the one by Joshua James](https://www.linuxcapable.com/how-to-install-latest-nginx-mainline-on-rocky-linux-8/) on [LinuxCapable](https://www.linuxcapable.com). Go give his site a read some time, there’s a lot of good stuff there. Now, on to this (beginner-friendly) guide:
+
+*Nginx* is a web server designed to be fast, efficient, and compatible with just about anything you can imagine. I personally use it a fair bit and—once you get the hang of it—it’s actually pretty easy to set up and configure. Here’s a short rundown of the standout features; Nginx is/has/can be:
+
+* A basic web server (one would hope)
+* A reverse proxy for directing traffic to multiple sites
+* A built-in load balancer for managing traffic to multiple websites
+* Built-in file caching for speed
+* WebSockets
+* FastCGI support
+* And, of course, IPv6
+
+It’s great! So just `sudo dnf install nginx`, right? Well, not exactly. **Rocky Linux repositories don’t have the latest production-ready version of Nginx.** Since our goal is bug-for bug compatibility with Red Hat Enterprise Linux, you can always ask them to update their repos. Or asking the *Nginx* people might work better (you’ll see what I mean).
+
+What *you* can do, right now, is install the “mainline” branch of Nginx yourself. It has all the latest updates and toys, and (to my mind) a simpler directory structure for its configuration files. Here’s how to see it all for yourself:
+
+!!! Note 
+
+    There's another branch called "stable", but it's actually a little outdated for most use cases. The "mainline" branch is considered by the Nginx developers to be thoroughly-tested and stable enough for everyone to use.
+
+## Prerequisites and Assumptions
+
+You’ll need:
+
+* An internet-connected Rocky Linux machine or server.
+* A basic familiarity with the command line.
+* The ability to run commands as root, either as the root user or with `sudo`.
+* A text editor of your choice, whether graphical or command-line based. For this tutorial, I’m using *nano*.
+
+## Installing the Repository
+
+This part isn’t quite as simple as installing an extra repository usually is. We’re going to have to create a custom repo file for *dnf* to use, and download *Nginx* from. Technically, we’re sort of repurposing  repositories for CentOS that were made and hosted by *Nginx* themselves. This solution may or may not be viable in the long term, but it’s working great for now.
```

Comment:
```suggestion
This part isn’t quite as simple as installing an extra repository usually is. We’re going to have to create a custom repo file for `dnf` to use, and download *Nginx* from. Technically, we’re sort of repurposing  repositories for CentOS that were made and hosted by *Nginx* themselves. This solution may or may not be viable in the long term, but it’s working great for now.
```

Reply:
I was just thinking about this: There is a module for nginx on Rocky for up-to version 1.20. Would this be better than a separate repository?

Reply:
I know I mentioned this to you on Mattermost @sspencerwire, but here's the reference for posterity. The Nginx module is the current "stable" branch, and it's fine, but the makers of Nginx say this (in an admittedly older blog post):

"In NGINX nomenclature, 'stable' means that no new features are added (the feature set is stable). Only major bug fixes are committed to that version.

"We develop new features and all bug fixes in the mainline branch. We operate a time‑based release process, so you can expect to see new mainline releases approximately once per month, with exceptional releases when necessary. Over the last year, the mainline has seen the introduction of SPDY 3.1 support, authentication via subrequests, SSL session ticket support, IPv6 support for DNS, and PROXY protocol support. We integrated community contributions for SSL support for uwsgi. We’ve extended error logging, added cache revalidation directives, added SMTP pipelining, added buffering options for FastCGI, improved support for MP4 streaming, and extended handling of byte‑range requests for streaming and caching.

"Note that stable does not mean more reliable or more bug‑free. In fact, the mainline is generally regarded as more reliable because we port all bug fixes to it, and not just critical fixes as for the stable branch. On the other hand, changes in the stable branch are very unlikely to affect third‑party modules. We don’t make the same commitment concerning the mainline, where new features can affect the operation of third‑party modules."

And:

"We recommend that in general you deploy the NGINX mainline branch at all times. The main reason to use the stable branch is that you are concerned about possible impacts of new features, such as incompatibility with third‑party modules or the inadvertent introduction of bugs in new features."

- https://www.nginx.com/blog/nginx-1-6-1-7-released/

I'm thinking that I'll explain this further in the guide, and show how to install from the module as well for people.

Reply:
For my part, I am still not familiar with the module system to be activated in dnf. But this is the official way.
In this case, I think there can be the "official" version and then an explanation "If you want to have an even newer version"... and keep the external repository and the actual explanation which is fine.
But the officially preferred version by redhat first.
That's good, it adds information to the page.

Translated with www.DeepL.com/Translator (free version)

Reply:
Well there's Red Hat official, and there's "getting all the latest security patches" official. I lean toward the latter, and I think I'll be saying as much in the guide.

Reply:
@EzequielBruni I'm fine with whatever you feel here. Rocky being a bug-for-bug compatible version of Red Hat means that the code is always going to lag a bit. It's by design. Doing things with your repository is absolutely fine. We do this all the time when we add in third-party repositories to get newer versions of software. Best example here is PHP 8.x. So you can add in the module install as an option, or not. Totally up to you.

Reply:
Oh I think it's genuinely a good idea to have both methods outlined for the sake of clarity and education. I like being thorough where I can.

And to think this all started because I realized we would need a full Nginx install guide to refer to for the LXD guide I'm writing.

Kind of curious to see what Antoine will have to say about that one. :)